src/controller_examples/rabbitmq_controller/proof/liveness/property.rs

```diff
@@ -0,0 +1,46 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: MIT
+#![allow(unused_imports)]
+use crate::external_api::spec::*;
+use crate::kubernetes_api_objects::{
+    api_method::*, common::*, config_map::*, dynamic::*, owner_reference::*, resource::*,
+    stateful_set::*,
+};
+use crate::kubernetes_cluster::spec::{
+    builtin_controllers::types::BuiltinControllerChoice,
+    cluster::*,
+    cluster_state_machine::Step,
+    controller::common::{ControllerActionInput, ControllerStep},
+    message::*,
+};
+use crate::rabbitmq_controller::{
+    common::*,
+    proof::resource::*,
+    resource::*,
+    spec::{reconciler::*, types::*},
+};
+use crate::temporal_logic::{defs::*, rules::*};
+use vstd::prelude::*;
+
+verus! {
+
+pub type RMQCluster = Cluster<RabbitmqClusterView, EmptyAPI, RabbitmqReconciler>;
+
+spec fn liveness(rabbitmq: RabbitmqClusterView) -> TempPred<RMQCluster>
+    recommends
+        rabbitmq.well_formed(),
+{
+    always(lift_state(RMQCluster::desired_state_is(rabbitmq))).leads_to(always(lift_state(current_state_matches(rabbitmq))))
+}
+
+pub open spec fn current_state_matches(rabbitmq: RabbitmqClusterView) -> StatePred<RMQCluster> {
+    |s: RMQCluster| {
+        forall |kind: ResourceKind| {
+            let key = (#[trigger] get_request_view(kind, rabbitmq)).key;
+            &&& s.resources().contains_key(key)
+            &&& resource_state_matches(kind, rabbitmq, s.resources()[key])
```

Comment:
@marshtompsxd , I specify the property here. And you can go to `rabbitmq_controller::resource` module for the separate spec for different resources. I guess the one for statefulset may be incorrect.

Reply:
> I guess the one for statefulset may be incorrect.

Why so?

Reply:
now i see